### PR TITLE
[4HrPR2Rm] Use internal function instead of procedure for compatibility check.

### PR DESCRIPTION
**Pull request by Lojjs**
This has been requested by Neo4j Aura, which couldn't see if the procedure call came from APOC or a user

### CODE REVIEW

File: core/src/test/java/apoc/cypher/CypherIsVersionDifferentTest.java [hunk 1]
```diff
@@ -12,30 +10,33 @@ public class CypherIsVersionDifferentTest {
...
-        assertTrue(isVersionDifferent(List.of("5.1.0"), "5.2.0"));
+        assertTrue(isVersionDifferent("3.5", "4.4.0.2"));
+        assertTrue(isVersionDifferent("5_0", "4.4.0.2"));
+        assertFalse(isVersionDifferent("3.5.12", "3.5.0.9"));
+        assertFalse(isVersionDifferent("3.5.12", "3.5.1.9"));
+        assertFalse(isVersionDifferent("4.4.5", "4.4.0.4"));
+        assertFalse(isVersionDifferent("5.1.1", "5.1.0"));
+        assertTrue(isVersionDifferent("5.1.0", "5.2.0"));
+        assertFalse(isVersionDifferent("5.1-aura", "5.1.0"));
+        assertTrue(isVersionDifferent("4.4-aura", "5.1.0"));
```
Comment (nadja-muller) [inline core/src/test/java/apoc/cypher/CypherIsVersionDifferentTest.java:21]: Is that correct? :sweat_smile:

Comment (Lojjs) [reply core/src/test/java/apoc/cypher/CypherIsVersionDifferentTest.java:21]: Yes I think so. The method checks if the versions are **not** compatible. So this checks that Neo4j 5.1.x and 5.1-aura are compatible with apoc 5.1.x while the combos Neo4j 5.1.x/APOC 5.2.x resp. Neo4j 4.4-aura/APOC 5.1.x are not compatible

Comment (nadja-muller) [reply core/src/test/java/apoc/cypher/CypherIsVersionDifferentTest.java:21]: I completely missed the assertTrue/asertFalse difference. Thank you for the explanation :+1:

**Review by nadja-muller — APPROVED** [review]
Looks like there was a flaky test failure in the CI, but otherwise it LGTM :smile:

**Pull request merged**